src/Gaffer/MessageHandler.cpp

```diff
@@ -0,0 +1,104 @@
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of Image Engine Design Inc nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/MessageHandler.h"
+
+#include "Gaffer/Context.h"
+#include "Gaffer/Plug.h"
+#include "Gaffer/Process.h"
+
+#include "IECore/FilteredMessageHandler.h"
+#include "IECore/VectorTypedData.h"
+
+#include "boost/algorithm/string/join.hpp"
+#include "boost/range/adaptor/transformed.hpp"
+
+#include <sstream>
+
+using namespace stdl
+
+using boost::adaptors::transformed;
+using boost::algorithm::join;
+
+namespace
+{
+	IECore::InternedString g_scenePath("scene:path");
+	IECore::InternedString g_frame( "frame" );
+};
+
+namespace Gaffer
+{
+
+class MessageHandler : public IECore::FilteredMessageHandler
+{
+	public:
+
+		IE_CORE_DECLAREMEMBERPTR( MessageHandler );
+
+		MessageHandler( IECore::MessageHandlerPtr handler ) : IECore::FilteredMessageHandler( handler )
+		{
+		}
+
+		~MessageHandler() override
+		{
+		}
+
+		void handle( Level level, const string &context, const string &message ) override
+		{
+			if( const Process *p = Process::current() )
+			{
+				stringstream ss;
+				ss << context;
+
+				ss << " [ plug: '" << p->plug()->fullName() << "' path: '";
+				auto path = p->context()->get<vector<IECore::InternedString> >( g_scenePath, vector<IECore::InternedString>() );
+				ss << boost::algorithm::join( path | transformed([](const IECore::InternedString& s) -> const string& { return ( const string& ) s;}), ".");
+				ss << "' time:" << p->context()->get<float>(g_frame, 0.0f) << " ]";
+				m_handler->handle( level, ss.str(), message );
```

Comment:
I thought we'd agreed that we'd always output the original message unchanged, and then follow up with a separate  DEBUG message containing this information. Is there any reason you didn't do that?

Reply:
I thought the messages looked unrelated in the output. Implemented what was agreed in a21c00b

Reply:
In practice I'm not sure if the messages look connected and that might be confusing to some users.

Reply:
I wonder if outputting as "DEBUG : Gaffer::Process for previous message : ..." would help?